src/software/firmware/srcs/respirator.cpp

```diff
@@ -217,4 +218,50 @@ void loop() {
     }
 }
 
+uint32_t stoppedCentiSec = 0;
+
+void runStoppedCycle() {
+    if (activationController.centiSecWhileStopped() == 0) {
+        // TODO perform all the necessary reset tasks
+        // Reset display
+        // Reset alarms
+        // Reset valves?
+    }
+
+    // Just wait one compute period.
+    uint32_t lastpControllerComputeDate = 0uL;
+    while (true) {
+        uint32_t currentDate = millis();
+        if ((currentDate - lastpControllerComputeDate) >= PCONTROLLER_COMPUTE_PERIOD) {
+            break;
+        }
+        lastpControllerComputeDate = currentDate;
+        IWatchdog.reload();
+    }
```

Comment:
Is there a possiblility that (currentDate - lastpControllerComputeDate) >= PCONTROLLER_COMPUTE_PERIOD) stays false. Then updating lastpControllerComputeDate with currentDate in the next line will create a non return condition continuously kipping stuck in the while(true). 
IWatchdog keeps reloading and no button can be triggered in that case the device needs to be power off/on to force reboot, right?

Reply:
I kept the logic used in the main respiratory cycle loop. I dont know what is the risk that millis() returns incoherent data.

Reply:
Yes I understand the logic. millis() is returning the number of milliseconds passed since the program started which is fine and can last for 50 days.
I am afraid that the loop will loop forever with the statement you use after lastpControllerComputeDate = currentDate; because it won't reach (currentDate - lastpControllerComputeDate) >= PCONTROLLER_COMPUTE_PERIOD (//10ms here)
If it is the behaviour you want, than it is fine to me.

Reply:
@ThomasL01 you're right. The loop looped forever. I change the behaviour and merged both methods. 
We want to avoid duplication of some part like watchdog reload, keyboardLoop and batteryLoop.